src/arch/x86_64/fdt.rs

```diff
@@ -0,0 +1,62 @@
+use alloc::vec::Vec;
+
+use multiboot::information::{MemoryType, Multiboot};
+use vm_fdt::{Error as FdtError, FdtWriter};
+
+use super::{mb_info, MEM};
+
+pub struct DeviceTree(Vec<u8>);
```

Comment:
I think this does not to contain data. Instead, we could have a freestanding associated function:

```rust
pub struct DeviceTree(());

impl DeviceTree {
    #[cfg(all(target_os = "none", not(feature = "fc")))]
    fn new() -> Result<&'static [u8], FdtError> {
        // ...
        let fdt = fdt.finish()?;
        Ok(fdt.leak())
    }
}
```

This would also avoid the need for a new `static` item.